cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -760,12 +768,16 @@ public MapAndStatus(T loadMap, K status) {
                                     vertx.cancelTimer(t);
                                     p.complete(buildRebalanceStatus(null, KafkaRebalanceState.Stopped, validate(reconciliation, currentKafkaRebalance)));
                                 } else {
-
                                     requestRebalance(reconciliation, host, apiClient, kafkaRebalance, true, rebalanceOptionsBuilder,
                                             currentKafkaRebalance.getStatus().getSessionId())
                                         .onSuccess(rebalanceMapAndStatus -> {
                                             // If the returned status has an optimization result then the rebalance proposal
                                             // is ready, so stop the polling
+                                            KafkaRebalanceStatus status = rebalanceMapAndStatus.getStatus();
+                                            if (rebalanceAnnotation(reconciliation, currentKafkaRebalance) != null) {
```

Comment:
aren't we trying to always return an enum (and not `null`) from the `rebalanceAnnotation` method? what would `null` mean?